src/funnels/Depositor.sol

```diff
@@ -0,0 +1,301 @@
+// SPDX-FileCopyrightText: © 2020 Lev Livnev <lev@liv.nev.org.uk>
+// SPDX-FileCopyrightText: © 2021 Dai Foundation <www.daifoundation.org>
+// SPDX-License-Identifier: AGPL-3.0-or-later
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.8.16;
+
+interface DepositedGemLike {
+    function approve(address, uint256) external;
+    function transfer(address, uint256) external;
+    function transferFrom(address, address, uint256) external;
+}
+
+
+// https://github.com/Uniswap/v3-periphery/blob/464a8a49611272f7349c970e0fadb7ec1d3c1086/contracts/interfaces/INonfungiblePositionManager.sol#L17
+interface PositionManagerLike {
```

Comment:
Need to develop the idea more, but I think the Depositor primitive should be as low level and generic as possible so that automated solutions can have as much flexibility as possible building on top of it.

So probably only enforcing `hop`, wrapping the needed pool functions, and making sure that the deposited/withdrawn amounts do not exceed pre-configured values.
Not sure the position management/wrapping should be in this level.

Reply:
Not sure about this anymore, resolving for now.